src/EditorFeatures/CSharpTest/QualifyMemberAccess/QualifyMemberAccessTests.cs

```diff
@@ -330,6 +330,67 @@ void M()
 CodeStyleOptions.QualifyFieldAccess);
         }
 
+        [WorkItem(40242, "https://github.com/dotnet/roslyn/issues/40242")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsQualifyMemberAccess)]
+        public async Task QualifyFieldAccess_Subpattern1()
+        {
+            await TestMissingAsyncWithOption(
+@"class Class
+{
+    int i;
+
+    void M(Class c)
+    {
+        if (c is { [|i|]: 1 })
```

Comment:
@zaytsev-victor @allisonchou can we add a test like `if (c is { i: [|i|] })`.  we can put a comment that it's ok that we don't support things here because it's not a legal pattern (because it is not const).